src/middlewared/middlewared/plugins/enclosure.py

```diff
@@ -497,23 +489,21 @@ def _set_model(self, data):
         elif R_SERIES_REGEX.match(self.encname) or R20A_REGEX.match(self.encname):
             self.model = self.product_name.replace("TRUENAS-", "")
             self.controller = True
-            if self.model in ["R20", "R20A"]:
-                self.model = f"{self.model}, Drawer #1"
         elif (
-            self.product_name in ["TRUENAS-R20", "TRUENAS-R20A"] and
             self.encname == "AHCI SGPIO Enclosure 2.00" and
-            len(data.splitlines()) == 6
+            self.system_info["system_product"] in ["TRUENAS-R20", "TRUENAS-R20A"]
```

Comment:
`self.system_info` does not exist so this will fail.
```suggestion
            self.product_name in ["TRUENAS-R20", "TRUENAS-R20A"]
```

Furthermore, we have a bug on line 541 and 543 because it also makes a reference to `system_info["system_product"]` which need to both be changed to `self.product_name`

Reply:
Should be resolved now